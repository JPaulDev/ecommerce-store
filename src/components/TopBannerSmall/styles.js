import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  display: flex;
  justify-content: space-around;
  min-height: 55px;
  position: relative;
`;

export const InnerContainer = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  column-gap: 15px;
  display: flex;
  justify-content: center;
  opacity: ${({ isActive }) => (isActive ? '1' : '0')};
  z-index: ${({ isActive }) => (isActive ? '10' : '0')};

  @media screen and (max-width: 999px) {
    min-width: 285px;
    transition: opacity 1s ease-in-out;
    position: absolute;
  }

  @media screen and (min-width: 1000px) {
    opacity: 1;
  }
`;

export const Image = styled.img`
  height: auto;
  width: ${({ displayWidth }) => displayWidth};
`;

export const Text = styled.span`
  color: var(--color-grey-2);
  font-size: 0.7rem;
`;

export const BoldText = styled.strong`
  display: block;
  font-size: 0.87rem;
`;
