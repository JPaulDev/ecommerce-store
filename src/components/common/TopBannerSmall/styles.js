import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  display: flex;
  justify-content: space-around;
  min-height: 55px;
  position: relative;
`;

export const InnerContainer = styled.div`
  align-items: center;
  background-color: var(--color-grey-6);
  column-gap: 15px;
  display: flex;
  justify-content: center;
  opacity: ${({ isActive }) => (isActive ? '1' : '0')};
  z-index: ${({ isActive }) => (isActive ? '10' : '0')};

  @media screen and (max-width: 999px) {
    min-width: 285px;
    transition: opacity 1s ease-in-out ${({ isActive }) => isActive && '1s'};
    position: absolute;
  }

  @media screen and (min-width: 1000px) {
    opacity: 1;
  }
`;

export const Text = styled.div`
  color: var(--color-grey-2);
  font-size: 0.7rem;

  strong {
    display: block;
    font-size: 0.87rem;
  }
`;
