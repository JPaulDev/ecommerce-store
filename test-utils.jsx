import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { setupStore } from './src/app/store';
import { UIProvider } from './src/contexts/UIContext';
import theme from './src/styles/theme';

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <UIProvider>{children}</UIProvider>
        </Provider>
      </ThemeProvider>
    );
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
export { renderWithProviders as render };
