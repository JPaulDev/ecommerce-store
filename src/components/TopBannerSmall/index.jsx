import { useState, useEffect } from 'react';
import useMediaQuery from '../../hooks/useMediaQuery';
import trophy from '../../assets/images/top-banner-small/trophy-logo.webp';
import trustpilot from '../../assets/images/top-banner-small/trustpilot-logo.webp';
import paypal from '../../assets/images/top-banner-small/paypal-logo.webp';
import * as Styled from './styles';

const content = [
  {
    image: trophy,
    width: 119,
    height: 55,
    displayWidth: '60px',
    boldText: 'Award winning',
    text: 'Systems and customer services',
  },
  {
    image: trustpilot,
    width: 121,
    height: 56,
    displayWidth: '73px',
    boldText: '4.8 stars on Trustpilot',
    text: 'From over 30,000 customer reviews',
  },
  {
    image: paypal,
    width: 167,
    height: 56,
    displayWidth: '80px',
    boldText: '0% interest up to 4 months',
    text: 'Minimum spend applies from Â£99',
  },
];

export default function TopBannerSmall() {
  const [slideIndex, setSlideIndex] = useState(1);
  const isMatch = useMediaQuery('(min-width: 1000px)');

  useEffect(() => {
    let timeout;
    const nextSlide = (slideIndex + 1) % content.length;

    if (!isMatch) {
      timeout = setTimeout(() => {
        setSlideIndex(nextSlide);
      }, 5000);
    }

    return () => clearTimeout(timeout);
  }, [isMatch, slideIndex]);

  return (
    <Styled.Container>
      {content.map((item, index) => (
        <Styled.InnerContainer
          key={item.text}
          isActive={index === slideIndex}
          data-testid={index}
        >
          <Styled.Image
            src={item.image}
            width={item.width}
            height={item.height}
            displayWidth={item.displayWidth}
            alt=""
          />
          <Styled.Text>
            <Styled.BoldText>{item.boldText}</Styled.BoldText>
            {item.text}
          </Styled.Text>
        </Styled.InnerContainer>
      ))}
    </Styled.Container>
  );
}
