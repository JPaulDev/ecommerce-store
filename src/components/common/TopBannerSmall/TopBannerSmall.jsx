import Image from 'next/future/image';
import { useState, useEffect } from 'react';
import trophy from '../../../../public/images/top-banner-small/trophy-logo.webp';
import trustpilot from '../../../../public/images/top-banner-small/trustpilot-logo.webp';
import paypal from '../../../../public/images/top-banner-small/paypal-logo.webp';
import useMediaQuery from '../../../lib/hooks/useMediaQuery';
import * as Styled from './styles';

const slides = [
  {
    image: trophy,
    boldText: 'Award winning',
    text: 'Systems and customer services',
  },
  {
    image: trustpilot,
    boldText: '4.8 stars on Trustpilot',
    text: 'From over 30,000 customer reviews',
  },
  {
    image: paypal,
    boldText: '0% interest up to 4 months',
    text: 'Minimum spend applies from Â£99',
  },
];

export default function TopBannerSmall() {
  const [slideIndex, setSlideIndex] = useState(1);
  const isMatch = useMediaQuery('(max-width: 999px)');

  useEffect(() => {
    let timeout;
    const nextSlide = (slideIndex + 1) % slides.length;

    if (isMatch) {
      timeout = setTimeout(() => {
        setSlideIndex(nextSlide);
      }, 5000);
    }

    return () => {
      clearTimeout(timeout);
    };
  }, [isMatch, slideIndex]);

  return (
    <Styled.Container>
      {slides.map((item, index) => (
        <Styled.InnerContainer
          key={item.text}
          isActive={index === slideIndex}
          data-testid={index}
        >
          <Image src={item.image} alt="" quality={100} />
          <Styled.Text>
            <strong>{item.boldText}</strong>
            {item.text}
          </Styled.Text>
        </Styled.InnerContainer>
      ))}
    </Styled.Container>
  );
}
