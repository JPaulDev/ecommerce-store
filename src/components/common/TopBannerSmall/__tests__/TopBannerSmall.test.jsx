import { act, render, screen } from '../../../../../test-utils';
import TopBannerSmall from '../index';

describe('TopBannerSmall', () => {
  it('should cycle through and display each slide after a set amount of time', () => {
    window.resizeTo(999, 769);
    jest.useFakeTimers();
    render(<TopBannerSmall />);

    expect(screen.getByTestId('0')).not.toBeVisible();
    expect(screen.getByTestId('1')).toBeVisible();
    expect(screen.getByTestId('2')).not.toBeVisible();

    act(() => {
      jest.runAllTimers();
    });

    expect(screen.getByTestId('0')).not.toBeVisible();
    expect(screen.getByTestId('1')).not.toBeVisible();
    expect(screen.getByTestId('2')).toBeVisible();
  });
});
