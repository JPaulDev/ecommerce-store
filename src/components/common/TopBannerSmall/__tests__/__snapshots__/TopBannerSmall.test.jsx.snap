// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBannerSmall component renders a banner with 3 icons and text next to each one 1`] = `
<div
  className="sc-a8ce1b99-0 eqxWUn"
>
  <div
    className="sc-a8ce1b99-1 YGGBa"
    data-testid={0}
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=100"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=100 1x, /_next/image?url=%2Fimg.jpg&w=48&q=100 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <div
      className="sc-a8ce1b99-2 jtLOmz"
    >
      <strong>
        Award winning
      </strong>
      Systems and customer services
    </div>
  </div>
  <div
    className="sc-a8ce1b99-1 iyqJuU"
    data-testid={1}
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=100"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=100 1x, /_next/image?url=%2Fimg.jpg&w=48&q=100 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <div
      className="sc-a8ce1b99-2 jtLOmz"
    >
      <strong>
        4.8 stars on Trustpilot
      </strong>
      From over 30,000 customer reviews
    </div>
  </div>
  <div
    className="sc-a8ce1b99-1 YGGBa"
    data-testid={2}
  >
    <img
      alt=""
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=100"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=100 1x, /_next/image?url=%2Fimg.jpg&w=48&q=100 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <div
      className="sc-a8ce1b99-2 jtLOmz"
    >
      <strong>
        0% interest up to 4 months
      </strong>
      Minimum spend applies from Â£99
    </div>
  </div>
</div>
`;
